daft/runners/partitioning.py

```diff
@@ -92,6 +92,36 @@ def from_table(cls, table: Table) -> PartitionMetadata:
 PartitionT = TypeVar("PartitionT")
 
 
+@runtime_checkable
```

Comment:
This is incompatible with Python 3.7, need to find a way to remove it

Reply:
I think you can do `from typing_extensions import runtime_checkable` for python 3.7